megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java

```diff
@@ -780,6 +781,91 @@ else if (numAdv > 1)
         addToTT("ArmorInternals", BR, entity.getTotalArmor(),
                 entity.getTotalInternal());
 
+        // Build a "status bar" visual representation of each
+        // component of the unit using block element characters.
+        if (GUIPreferences.getInstance().getBoolean(GUIPreferences.SHOW_ARMOR_MINIVIS_TT)) {
```

Comment:
Code-wise, this mostly looks fine. The only change I would recommend is splitting this functionality out to a separate method.